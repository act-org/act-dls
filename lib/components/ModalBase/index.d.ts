/**
 * @prettier
 */
import * as React from 'react';
import { ModalProps } from '@material-ui/core';
export declare type Props = ModalProps;
declare const ModalBase: React.FC<Props>;
export default ModalBase;
//# sourceMappingURL=index.d.ts.map