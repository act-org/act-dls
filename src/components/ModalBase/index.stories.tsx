/**
 * @prettier
 */

import * as React from 'react';
import { boolean } from '@storybook/addon-knobs';

import ModalBase from '.';

export const Base = (): React.ReactElement<any> => (
  <ModalBase
    disableBackdropClick
    disableEscapeKeyDown
    open={boolean('Open?', true)}
  >
    <div>Modal Content</div>
  </ModalBase>
);

export default {
  component: ModalBase,
  parameters: {
    info: {
      text: 'This is the base variant of the Modal component.',
    },
  },
  title: 'Atoms/Modal',
};
