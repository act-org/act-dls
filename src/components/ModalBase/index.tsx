/**
 * @prettier
 */

import * as React from 'react';
import { Modal, ModalProps } from '@material-ui/core';

export type Props = ModalProps;

const ModalBase: React.FC<Props> = (props: Props): React.ReactElement<any> => (
  <Modal {...props} />
);

export default ModalBase;
