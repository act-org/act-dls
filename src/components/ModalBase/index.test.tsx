/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import ModalBase from '.';

describe('ModalBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <ModalBase disableBackdropClick open>
        <div>Modal content</div>
      </ModalBase>,
    );

    expect(container).toMatchSnapshot();
  });
});
