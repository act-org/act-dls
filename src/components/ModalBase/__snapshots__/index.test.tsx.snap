// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBase matches the snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
